packages/blade/src/components/Button/_decisions/decisions.md

```diff
@@ -0,0 +1,81 @@
+# Button Decisions <!-- omit in toc -->
+- [Design](#design)
+- [Internal BaseButton Component](#internal-basebutton-component)
+- [Button Component](#button-component)
+- [Component Breakdown](#component-breakdown)
+- [Open Questions](#open-questions)
+
+## Design
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=5200%3A0) to all variants of the Button component
+
+## Internal BaseButton Component
+-  Internal component that exposes certain extra props to enable creation of Component Level Action Buttons like `positive`, `negative`, `neutral`
+-  **BaseButton API**:
+   -  **`variant`**: `primary` | `secondary` | `tertiary` | `link`
+      -  **default**: `primary`
+      -  Link will look like an inline Link component but will have the role of a `button`
+   -  **`intent`**: `positive` | `negative` | `notice` | `info` | `neutral`
+      -  **default**: `undefined`
+      -  We will use this to set the `intent` for internal Blade components that have positive, negative, notice, info, or neutral buttons like an `Alert` component
+   -  **`contrast`**: `high` | `low`
+      -  **default**: `low`
+      -  We need this for internal Blade components since a `positive` intent could have a `high` as well as a `low` contrast button
+   -  **`size`**: `large` | `medium` | `small` | `xsmall`
+      - **default**: `medium`
+   -  **`children`**: `string`
```

Comment:
I think we will be restricting it to string, because otherwise anyone can put any Icon or other components as prefix/suffix. 

// cc @chaitanyadeorukhkar

Reply:
Yup right. Makes sense. Can't think of any usecase where a nested element will be required inside a button 🤔

Reply:
If we are accepting `children` as string, that means it will basically be a text which we want to display on the button.

Rather than exposing `children`, why not expose `label` and `sub label`? What do you think?

Reply:
> Rather than exposing children, why not expose label and sub label? What do you think?

Yes, But at the end of the day it's a matter of API choices, both will work as long as we are consistent with the API.  

With children it's more intuitive and easy to teach people. Plus one less prop to worry about thus lesser API surface. (children can also be used as a prop but you get the point)

Reply:
I think there are genuine cases where we could need nested markup inside Button. Like suppose we need to show a pill inside button as shown in GitHub.

If we're saying that we'll have a rigid api and all these will be allowed only when design system supports a custom prop then ok.

<img width="593" alt="image" src="https://user-images.githubusercontent.com/83330415/169252726-666605be-fe5d-46f9-8fb4-2d453528948d.png">

Reply:
> If we're saying that we'll have a rigid api and all these will be allowed only when design system supports a custom prop then ok.

Yes, GitHub might have the use case, but it's depends on if our products have the use case, if it does we can look into making it more flexible. 

Saurav from the design side has audited components to collect most of the usecases.

Reply:
@aditya-razorpay we don't have any use-cases for such buttons as of now. We do have Buttons we icons and we intend to support that. Adding more accessories to button can be done later as and when we see more common patterns being introduced in our system

Reply:
@aditya-razorpay Nice observation! Given the nature of how GitHub as a product is it assigns numbers/counters as a composite component to many of its components like buttons, tabs, sidebar etc. Check [ref](https://primer.style/react/Button#counter-component). However, if you assess other commonly used design systems or even sites in Razorpay a pill inside a button is quite rare.